core/include/traccc/clusterization/impl/measurement_creation.ipp

```diff
@@ -92,38 +99,53 @@ TRACCC_HOST_DEVICE inline void fill_measurement(
     // [2] The Art of Computer Programming, Donald E. Knuth, second
     //     edition, chapter 4.2.2.
 
-    // A security check.
-    assert(cluster.empty() == false);
+    // Security checks.
+    assert(clusters.cell_indices()[index].empty() == false);
+    assert([&]() {
+        const unsigned int module_idx =
+            cells.module_index().at(clusters.cell_indices()[index].front());
+        for (const unsigned int cell_idx : clusters.cell_indices()[index]) {
+            if (cells.module_index().at(cell_idx) != module_idx) {
+                return false;
+            }
+        }
+        return true;
+    }() == true);
 
     // Calculate the cluster properties
     scalar totalWeight = 0.f;
     point2 mean{0.f, 0.f}, var{0.f, 0.f};
-    calc_cluster_properties(cluster, det_descr, mean, var, totalWeight);
+    calc_cluster_properties(index, cells, clusters, det_descr, mean, var,
+                            totalWeight);
 
     assert(totalWeight > 0.f);
 
     // Access the measurement in question.
     measurement& m = measurements[index];
 
     // The index of the module the cluster is on.
-    const cell::link_type module_link = cluster.at(0).module_link;
-
-    m.module_link = module_link;
-    m.surface_link = det_descr.geometry_id().at(module_link);
+    const unsigned int module_idx =
+        cells.module_index().at(clusters.cell_indices().at(index).front());
```

Comment:
Please use the proxy-based syntax which is more natural to read:

```suggestion
    const unsigned int module_idx =
        cells.at(clusters.at(index).cell_indices().front()).module_index();
```

Reply:
I really disagree. :thinking: When we only need a single variable, I much prefer to use the "container interface". Since that's the whole point of an SoA layout, that we could do this. Without creating temporary proxy objects all around the place.